XYZDrivers/src/main/java/com/xyzdrivers/repositories/ClaimsRepo.java

```diff
@@ -0,0 +1,44 @@
+package com.xyzdrivers.repositories;
+
+import com.xyzdrivers.models.Claim;
+import com.xyzdrivers.services.SQLService;
+import java.sql.SQLException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * DAO for claims model
+ * @author arthur
+ */
+public class ClaimsRepo {
+    
+    private SQLService sql;
+    
+    public ClaimsRepo(SQLService sql) {
+        this.sql = sql;
+    }
+    
+    /**
+     * Method to retrieve all of the claims in the database
+     * @return List<Claim> claims
+     */
+    public List<Claim> getAll() {
+        List<Object[]> results;
+        List<Claim> claims = new ArrayList<>();
+        try {
+            results = this.sql.retrieve("claims");
+            for (Object[] result : results) {
+                // TODO change LocalDate
+                Claim cl = new Claim(result[1].toString(), LocalDate.now(), result[3].toString(), result[4].toString(), Float.parseFloat(result[5].toString()));
+                claims.add(cl);
+            }
+        } catch (SQLException ex) {
+            Logger.getLogger(ClaimsRepo.class.getName()).log(Level.SEVERE, null, ex);
```

Comment:
The problem with this is the calling method has no idea an error occurred - up to you to leave as-is or not, since it will have no effect on the final product

Reply:
Hmmm good point. Not sure about this one then. If we catch here then we can just return an empty array and make the app a little reliant to error. But then if I throw the exception to the caller the exception can be passed through the call chain and handled at the top.I.e. this throws the error to processRequest in the controller which then throws it to the doGet method. But then the request would then fail as the controller would think that creating the connection throw the error, as its the same SqlException. I think to leave as is but if there is a more specific SqlQueryException we can use to distinguish then better to throw.

Reply:
It wouldn't improve the functionality of the overall program, so arguably it's not worth spending the effort to change. But if we're talking best practice, throwing is the ideal thing to do, since returning an empty list implies that there are genuinely no records in the database when this might not be true - and there's no way of distinguishing between the two scenarios.

We would just need to split the try...catch block up to tell whether creating the connection caused the error or the call to ClaimsRepo did it.

I've seen a few examples lying around where people catch the exception and re-throw it as a custom exception, e.g. MyPackageException. Another possible option but in my opinion a bit over-the-top and better suited to libraries.